Crash report 8.9: NPE in ThemeBrowserActivity.setThemeBrowserFragment

On a Jetpack or wpcom site, when I tap the Themes button, I get this crash:

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android.beta/org.wordpress.android.ui.themes.ThemeBrowserActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void org.wordpress.android.ui.themes.ThemeBrowserActivity.setThemeBrowserFragment(org.wordpress.android.ui.themes.ThemeBrowserFragment)' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
	at android.app.ActivityThread.access$800(ActivityThread.java:144)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5223)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void org.wordpress.android.ui.themes.ThemeBrowserActivity.setThemeBrowserFragment(org.wordpress.android.ui.themes.ThemeBrowserFragment)' on a null object reference
	at org.wordpress.android.ui.themes.ThemeBrowserFragment.onActivityCreated(ThemeBrowserFragment.java:139)
	at android.app.Fragment.performActivityCreated(Fragment.java:2061)
	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:912)
	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
	at android.app.BackStackRecord.run(BackStackRecord.java:833)
	at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1452)
	at android.app.Activity.performStart(Activity.java:5948)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2261)
	... 10 more
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.java
@@ -109,6 +109,8 @@ public class ThemeBrowserFragment extends Fragment implements RecyclerListener {
     public void onAttach(Activity activity) {
         super.onAttach(activity);
 
+        // This version of onAttach will be called for all Android versions but it's deprecated, so we only want to
+        // use it when we have to. https://github.com/wordpress-mobile/WordPress-Android/issues/6937
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
             attachActivity((ThemeBrowserActivity) activity);
         }

--- a/WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserFragment.java
@@ -1,11 +1,13 @@
 package org.wordpress.android.ui.themes;
 
 import android.annotation.SuppressLint;
+import android.app.Activity;
 import android.app.Fragment;
 import android.content.Context;
 import android.database.Cursor;
 import android.database.MatrixCursor;
 import android.database.MergeCursor;
+import android.os.Build;
 import android.os.Bundle;
 import android.text.TextUtils;
 import android.view.LayoutInflater;
@@ -100,11 +102,24 @@ public class ThemeBrowserFragment extends Fragment implements RecyclerListener {
     public void onAttach(Context context) {
         super.onAttach(context);
 
+        attachActivity((ThemeBrowserActivity) context);
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            attachActivity((ThemeBrowserActivity) activity);
+        }
+    }
+
+    private void attachActivity(ThemeBrowserActivity activity) {
         try {
-            mCallback = (ThemeBrowserFragmentCallback) context;
-            mThemeBrowserActivity = (ThemeBrowserActivity) context;
+            mCallback = activity;
+            mThemeBrowserActivity = activity;
         } catch (ClassCastException e) {
-            throw new ClassCastException(context.toString() + " must implement ThemeBrowserFragmentCallback");
+            throw new ClassCastException(activity.toString() + " must implement ThemeBrowserFragmentCallback");
         }
     }
 

